Targets/Mashow/Sources/DrinkSelection/View/DrinkSelectionViewController.swift

```diff
@@ -0,0 +1,287 @@
+//
+//  DrinkSelectionViewController.swift
+//  Mashow
+//
+//  Created by ZENA on 8/3/24.
+//  Copyright © 2024 com.alcoholers. All rights reserved.
+//
+
+import UIKit
+import Combine
+import SnapKit
+
+final class DrinkSelectionViewController: UIViewController {
+    
+    init(viewModel: DrinkSelectionViewModel!) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    let viewModel: DrinkSelectionViewModel!
+    private var cancellables = Set<AnyCancellable>()
+    private let drinkTypeList = DrinkSelectionViewModel.DrinkType.allCases
+    
+    private var removedButtonTag: Int?
+    typealias TypeButtonTag = Int
+    private var addedDrinkTypeButtons = [TypeButtonTag: UIButton]()
```

Comment:
```suggestion
    private var removedButtonTag: DrinkType?
    private var addedDrinkTypeButtons = [DrinkType: UIButton]()
```
요게 좀 더 가시적일 것 같은데 어떠세용..? 급한 건 아니라 이건 치워놓고 다음에 해도 됩니다.

Reply:
대신 저렇게 하면 이 부분이 문제가 될텐데
``` Swift
@objc private func didTapToRemoveType(_ sender: UIButton) {
    removedButtonTag = sender.tag // <- 여기
    guard let index = drinkTypeList.firstIndex(where: { $0 == removedButtonTag }) else { return }
    viewModel.removeType(drinkTypeList[index])
}
```

이렇게 쓰면 해결은 될 것 같은데
``` Swift
class DrinkTypeButton: UIButton {
    var drinkType: DrinkType
    init(_ drinkType: DrinkType) {
        self.drinkType = type
        super.init(frame: .zero)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}

...

@objc private func didTapToRemoveType(_ sender: UIButton) {
    guard let sender = sender as? DrinkTypeButton else { return }
    removedButtonType = sender.drinkType // <- 여기
    guard let index = drinkTypeList.firstIndex(where: { $0 == removedType }) else { return }
    viewModel.removeType(drinkTypeList)
}
```

작업이 커질 것 같으니까 일단 스킵하시죠ㅎ

Reply:
#48